tinygrad/runtime/ops_metal.py

```diff
@@ -10,32 +9,37 @@
 METAL_XCODE = getenv("METAL_XCODE")
 
 class _METAL:
-  def __init__(self):
+  def __init__(self) -> None:
     self.mtl_buffers_in_flight: List[Any] = []
     self.device = Metal.MTLCreateSystemDefaultDevice()
     self.mtl_queue = self.device.newCommandQueue()
   # TODO: is there a better way to do this?
-  def synchronize(self):
+  def synchronize(self) -> None:
     for cbuf in self.mtl_buffers_in_flight: cbuf.waitUntilCompleted()
     self.mtl_buffers_in_flight.clear()
 METAL = _METAL()
 
 class RawMetalBuffer(RawBufferMapped):
-  def __init__(self, size:int, dtype:DType): super().__init__(size, dtype, METAL.device.newBufferWithLength_options_(size*dtype.itemsize, Metal.MTLResourceStorageModeShared))
-  def __del__(self):
-    self._buf.release()
+  def __init__(self, size:int, dtype:DType) -> None:
+    super().__init__(size, dtype, METAL.device.newBufferWithBytesNoCopy_length_options_deallocator_(None, size * dtype.itemsize, Metal.MTLResourceStorageModeShared, None))
+  def __del__(self) -> None:
+    self.release()
     super().__del__()
+  def release(self) -> None:
+    purgeable_state = 2  # 2 represents the MTLPurgeableState.Empty state
```

Comment:
On this webapge, it says that "2" is "nonvolatile", is it different through the py-objc framework? https://developer.apple.com/documentation/metal/mtlpurgeablestate/mtlpurgeablestatenonvolatile